src/mds/CDir.cc

```diff
@@ -656,6 +656,14 @@ void CDir::try_remove_unlinked_dn(CDentry *dn)
 void CDir::unlink_inode_work(CDentry *dn)
 {
   CInode *in = dn->get_linkage()->get_inode();
+  if (in) {
+    for (auto fin_set : in->batch_ops[CEPH_MDS_OP_LOOKUP]) {
+      for (auto fin : fin_set.second) {
+  	fin->complete(-ENOENT);
+      }
+    }
+    in->batch_ops[CEPH_MDS_OP_LOOKUP].clear();
+  }
```

Comment:
I think this is not safe for inode with multiple linkages

Reply:
I don't see how do you resolve this issue

Reply:
@ukernel Hi, thanks for your review:-)
But I don't quite follow you. I think:
1. Without the optimization, if the second "touch file" gets processed after "rm -f file" and the second "stat file" gets processed between "rm -f file" and "touch file", it would get a -ENOENT, too;
2. With the optimization, if the second "touch file" gets processed after "rm -f file" and the second "stat file" gets procesed, together with the first one, after "touch file" or the second "stat file" gets processed after the first one and "touch file", it can also succeed.
Am I right about this? Thanks:-)